### DISCUSSION TITLE
Motion Manager in dependency

### DISCUSSION THREAD

**Discussion by rpatel8del** [pointfreeco/swift-composable-architecture#1931] [Q&A]
I'm adding a dependency for the CMMotionManager, I need the updates that reflect in some UI changes but for some reason this results in a laggy UI. Am I doing something wrong here?
 
```
public extension MotionManagerClient {
    static var live: Self {
        let manager = MotionManager(motionManager: CMMotionManager())
        return MotionManagerClient(startUpdates: { await manager.startUpdates() })
    }
}

private actor MotionManager {
    private let motionManager: CMMotionManager
    init(motionManager: CMMotionManager) {
        self.motionManager = motionManager
    }
    func startUpdates() -> AsyncStream<CGSize> {
        AsyncStream { continuation in
            if !motionManager.isDeviceMotionAvailable {
                continuation.yield(.zero)
            }
            motionManager.startDeviceMotionUpdates(to: .main) { data, error in
                if let data = data {
                    continuation.yield(.init(width: data.attitude.roll,
                                             height: data.attitude.pitch))
                }
            } 
            continuation.onTermination =
              { @Sendable _ in
                  self.motionManager.stopDeviceMotionUpdates()
              }
        }
    }
}
```

**Comment by Alex293** [comment]
Be sure to scope only those changes when you use it because it will make you view render excessively. You can also try to denounce small increments in values. I haven’t tried using motion manager with tca yet but I plan to do so soon

**Reply by rpatel8del** [reply]
Is there a pattern for this or is it just the usual scope down the state for the specific view and then update that state using the motion manager?

**Reply by Alex293** [reply]
I was thinking about the usual pattern, ensure your parents views are not refreshing because of a too large view store scoping somewhere. If it’s not that, maybe your changes using the motion values update the frame or something similar and cause chain redraws of other views ?
